modules/jms11/src/test/java/org/apache/ignite/stream/jms11/IgniteJmsStreamerTest.java

```diff
@@ -537,29 +539,32 @@ public void testExceptionListener() throws Exception {
                     System.out.println("ERROR");
 
                     lsnrExceptions.add(e);
+
+                    latch.countDown();
                 }
             });
 
             jmsStreamer.setDestination(dest);
 
-            try {
-                jmsStreamer.start();
+            GridTestUtils.assertThrowsWithCause(new Callable<Void>() {
+                @Override public Void call() throws Exception {
+                    jmsStreamer.start();
 
-                fail("Expecting security exception");
-            } catch (IgniteException ie) {
-                assertTrue("Caught IgniteException, expected: ", ie.getCause().getCause() instanceof SecurityException);
-            }
+                    return null;
+                }
+            }, SecurityException.class);
 
-            while(lsnrExceptions.isEmpty())
-                Thread.sleep(200);
+            latch.await(5, TimeUnit.SECONDS);
```

Comment:
Better to wrap `await()` calls into `assertTrue()` to make sure that the test fails if the `await()` times out.

Reply:
Added that.